docs/paper/dev/api/registries.mdx

```diff
@@ -0,0 +1,165 @@
+---
+slug: /dev/registries
+description: A guide to registries and their modification on paper
+---
+
+# Registries
+
+:::danger[Experimental]
+The Registry API and anything that uses it is currently experimental and may change in the future.
+:::
+
+## What is a registry ?
+
+In the context of Minecraft, a registry holds onto a set of values of the same type, identifying
+each by a key. An example of such a registry would be the <Javadoc name={"org.bukkit.Registry#ITEM"}>ItemType registry</Javadoc> which holds all known item types.
+Registries are available via the <Javadoc name={"io.papermc.paper.registry.RegistryAccess"}>RegistryAccess</Javadoc>.
+
+While a large portion of registries are defined by the server and client independently, more and
+more are defined by the server and sent to the client while joining the server.
+This enables the server, and to that extend plugins, to define custom content for both itself and
+clients playing on it.
+Notable examples include **enchantments** and **biomes**.
+
+### Retrieving values from a registry
+
+To retrieve elements from a registry, their respective keys can be used.
+The API defines two types of keys.
+<ol>
+  <li>
+    `net.kyori.adventure.key.Key` represents a namespace and a key.
+  </li>
+  <li>
+    <Javadoc name={"io.papermc.paper.registry.TypedKey"}>TypedKey</Javadoc> wraps an adventure key,
+    but also includes the <Javadoc name={"io.papermc.paper.registry.TypedKey#registryKey()"}>key of
+    the registry</Javadoc> the
+    <Javadoc name={"io.papermc.paper.registry.TypedKey"}>TypedKey</Javadoc> belongs to.
+  </li>
+</ol>
+An example of retrieving the sharpness enchantment using
+<Javadoc name={"io.papermc.paper.registry.TypedKey"}>TypedKeys</Javadoc> looks as follows:
+
+```java
+// Fetch the enchantment registry from the registry access
+final Registry<Enchantment> enchantmentRegistry = RegistryAccess
+    .registryAccess()
+    .getRegistry(RegistryKey.ENCHANTMENT);
+
+// Get the sharpness enchantment using its key.
+// getOrThrow may be replaced with get if the registry may not contain said value
+final Enchantment enchantment = enchantmentRegistry.getOrThrow(TypedKey.create(
+    RegistryKey.ENCHANTMENT, Key.key("minecraft:sharpness"))
+);
+
+// Same as above, but using generated typed keys.
+// Only Vanilla entries have generated keys, for custom entries, the above method must be used.
+final Enchantment enchantment = enchantmentRegistry.getOrThrow(EnchantmentKeys.SHARPNESS);
+```
+
+### Referencing registry values
+
+Referencing entries in a registry is easier said then done.
+While for most cases a plain Collection of the values might suffice, alternative approaches are
```

Comment:
presuming it's a reference to the Java stdlib class, `Collection` could be encased in backticks and/or linked to the corresponding Javadoc